tests/system-defaults-write.nix

```diff
@@ -1,13 +1,69 @@
 { config, pkgs, ... }:
 
 {
+  system.defaults.NSGlobalDomain.AppleFontSmoothing = 1;
+  system.defaults.NSGlobalDomain.AppleKeyboardUIMode = 3;
+  system.defaults.NSGlobalDomain.ApplePressAndHoldEnabled = true;
+  system.defaults.NSGlobalDomain.AppleShowAllExtensions = true;
+  system.defaults.NSGlobalDomain.AppleShowScrollBars = "Always";
+  system.defaults.NSGlobalDomain.NSAutomaticCapitalizationEnabled = false;
+  system.defaults.NSGlobalDomain.NSAutomaticDashSubstitutionEnabled = false;
+  system.defaults.NSGlobalDomain.NSAutomaticPeriodSubstitutionEnabled = false;
+  system.defaults.NSGlobalDomain.NSAutomaticQuoteSubstitutionEnabled = false;
+  system.defaults.NSGlobalDomain.NSAutomaticSpellingCorrectionEnabled = false;
+  system.defaults.NSGlobalDomain.NSDisableAutomaticTermination = true;
+  system.defaults.NSGlobalDomain.NSDocumentSaveNewDocumentsToCloud = false;
+  system.defaults.NSGlobalDomain.NSNavPanelExpandedStateForSaveMode = true;
+  system.defaults.NSGlobalDomain.NSNavPanelExpandedStateForSaveMode2 = true;
+  system.defaults.NSGlobalDomain.NSTableViewDefaultSizeMode = 2;
+  system.defaults.NSGlobalDomain.NSTextShowsControlCharacters = true;
+  system.defaults.NSGlobalDomain.NSUseAnimatedFocusRing = false;
+  system.defaults.NSGlobalDomain.NSScrollAnimationEnabled = true;
+  system.defaults.NSGlobalDomain.NSWindowResizeTime = "0.01";
+  system.defaults.NSGlobalDomain.InitialKeyRepeat = 10;
   system.defaults.NSGlobalDomain.KeyRepeat = 1;
+  system.defaults.NSGlobalDomain.PMPrintingExpandedStateForPrint = true;
+  system.defaults.NSGlobalDomain.PMPrintingExpandedStateForPrint2 = true;
+  system.defaults.NSGlobalDomain."com.apple.mouse.tapBehavior" = 1;
+  system.defaults.NSGlobalDomain."com.apple.trackpad.enableSecondaryClick" = true;
+  system.defaults.NSGlobalDomain."com.apple.trackpad.trackpadCornerClickBehavior" = 1;
+  system.defaults.NSGlobalDomain."com.apple.springing.enabled" = true;
+  system.defaults.NSGlobalDomain."com.apple.springing.delay" = "0.0";
+  system.defaults.NSGlobalDomain."com.apple.swipescrolldirection" = true;
   system.defaults.dock.autohide-delay = "0.24";
   system.defaults.dock.orientation = "left";
 
   test = ''
     echo checking defaults write in /activate-user >&2
+    grep "defaults write -g 'AppleFontSmoothing' -int 1" ${config.out}/activate-user
```

Comment:
This is probably a bit overkill, but evaluating everything with a value will help make sure nothing is broken.

Reply:
this. names/capitalization are easy to get wrong, variable selection + regression testing for the future. Since they run so quickly and the output is well-defined and not in flux, I don't believe it's bad practice to cover as much ground as possible.